server/server.go

```diff
@@ -1073,12 +1076,37 @@ func (s *Server) handleBodyAsMessageAutoDetect(m *message, body *util.PeekedRead
 	return nil
 }
 
-func (s *Server) handleBodyAsTextMessage(m *message, body *util.PeekedReadCloser) error {
+func (s *Server) handleBodyAsTextMessage(m *message, body *util.PeekedReadCloser, template bool) error {
 	if !utf8.Valid(body.PeekedBytes) {
 		return errHTTPBadRequestMessageNotUTF8.With(m)
 	}
 	if len(body.PeekedBytes) > 0 { // Empty body should not override message (publish via GET!)
-		m.Message = strings.TrimSpace(string(body.PeekedBytes)) // Truncates the message to the peek limit if required
+		peakedBody := strings.TrimSpace(string(body.PeekedBytes)) // Truncates the message to the peek limit if required
+		if template && gjson.Valid(peakedBody) {
+			// Replace JSON paths in message
+			r := regexp.MustCompile(`\${([^}]+)}`)
```

Comment:
move to global var at the top of the file; don't need to re-compile the same regex over and over again

```
var (
   templateVarRegex = regexp.MustCompile(...
)
```